src/pip/_internal/utils/logging.py

```diff
@@ -44,15 +44,23 @@ def get_indentation():
 
 
 class IndentingFormatter(logging.Formatter):
+    def __init__(self, *args, **kwargs):
+        self.timestamp = kwargs.pop("timestamp", False)
+        super(IndentingFormatter, self).__init__(*args, **kwargs)
 
     def format(self, record):
         """
         Calls the standard formatter, but will indent all of the log messages
         by our current indentation level.
         """
         formatted = logging.Formatter.format(self, record)
+        prefix = ''
+        if self.timestamp:
+            prefix = logging.Formatter.formatTime(
```

Comment:
I think it would be better to try using `super()` here as that is preferred. (I'm not sure why `super()` isn't being used above.)